models/refresh_token.go

```diff
@@ -100,6 +100,18 @@ func GetValidChildToken(tx *storage.Connection, token *RefreshToken) (*RefreshTo
 	return refreshToken, nil
 }
 
+func FindTokenBySessionID(tx *storage.Connection, sessionId *uuid.UUID) (*RefreshToken, error) {
+	refreshToken := &RefreshToken{}
+	err := tx.Q().Where("session_id = ?", sessionId).Order("created_at asc").First(refreshToken)
```

Comment:
i don't think we have an index on `session_id` - might be worth adding one because this table is usually pretty large and having an index would cut down on the filtering time

Reply:
will add an index

Reply:
will also file a pr with the index against `master` once this goes through